### ISSUE TITLE
Incorrect nullable flow analysis in recursive local function calls

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#45934]
**Version Used**: VS 16.7.0 preview 3

**Steps to Reproduce**:

compile the following:
```csharp
using System;
#nullable enable
public class C {
    public void M() {
        string? x = null;
        x = "";
        N2();
        void N1() {
            N2();
            _ = x.ToString();
            
        }
        void N2() {
            Inner();
            void Inner()
            {
                N1();
                _ = x.ToString();
            }
        }
    }
}
```

**Expected Behavior**: no warning

**Actual Behavior**: "warning CS8602: Dereference of a possibly null reference." on both uses of `x.ToString()`

**Comment by RikkiGibson** [comment]
I believe the root cause of this issue is the same as what is described in https://github.com/dotnet/roslyn/issues/40157#issuecomment-655056976

**Comment by YairHalberstadt** [comment]
@RikkiGibson here all local functions are used

**Comment by RikkiGibson** [comment]
If a local function is only used within local functions we basically fail to thread updated nullable state into it. This is definitely in the bucket of “things that need improvement about local function analysis”.

**Comment by RikkiGibson** [comment]
This is something that we should solve by simply doing another pass if we see a local function usage after we have visited the local function declaration.

**Comment by RikkiGibson** [comment]
Closing as a dupe of #44805.
